frontend/src/metabase/dashboard/actions.js

```diff
@@ -544,6 +544,8 @@ export const fetchCardData = createThunkAction(FETCH_CARD_DATA, function(
     // If the dataset_query was filtered then we don't have permisison to view this card, so
     // shortcircuit and return a fake 403
     if (!card.dataset_query) {
+      // introduce delay to avoid setting this request complete before it was set to pending
+      await delay(1);
```

Comment:
Can we see if setting the loadingIds [here](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/dashboard/reducers.js#L304) will put everything in the right order? Then FETCH_DASHBOARD_CARD_DATA can be responsible for setting the start time and loading status. My concern is that 1ms may work in this scenario, but we could run into others where that still wouldn't be fast enough (such as a dashboard with many cards)